contracts/policy-agents/PolicyOfferExecutor.sol

```diff
@@ -0,0 +1,129 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../interfaces/IClaimsManager.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract PolicyOfferExecutor {
+    using ECDSA for bytes32;
+    using SafeERC20 for IERC20;
+
+    address public immutable claimsManager;
+    address public immutable offerSigner;
+    address public immutable token;
+    address public immutable beneficiary;
+
+    constructor(
+        address _claimsManager,
+        address _offerSigner,
+        address _token,
+        address _beneficiary
+    ) {
+        claimsManager = _claimsManager;
+        offerSigner = _offerSigner;
+        token = _token;
+        beneficiary = _beneficiary;
+    }
+
+    function executeOffer(
+        bytes calldata policyData,
+        uint256 offerAmount,
+        uint256 offerExpiration,
+        bytes calldata offerSignature
+    ) external returns (bytes32 policyHash) {
+        require(block.timestamp < offerExpiration, "Offer expired");
```

Comment:
Should we have a MIN_EXPIRATION_DURATION constant to prevent caller from creating a policy that might expire  in a sec or some other weird value by accident?

Reply:
This is the offer expiration, not the policy expiration. We can't know how long was the expiration duration was set to be while signing, so I don't understand this.

Reply:
Right, my bad. I got confused with `claimsAllowedUntil` argument 🤦🏻